browser-extension/common/src/popup/components/SessionRecording/SessionRecordingView.tsx

```diff
@@ -1,39 +1,43 @@
-import React, { useEffect, useState } from "react";
-import { Button, Col, Divider, Row, Typography } from "antd";
+import React, { useCallback, useEffect, useState } from "react";
+import { Button, Col, Row, Typography } from "antd";
 import config from "../../../config";
 import { CLIENT_MESSAGES } from "../../../constants";
 import VideoRecorderIcon from "../../../../resources/icons/videoRecorder.svg";
-import "./sessionRecordingView.css";
+import { PlayCircleFilled, SaveFilled } from "@ant-design/icons";
 import { EVENT, sendEvent } from "../../events";
+import "./sessionRecordingView.css";
 
 const SessionRecordingView: React.FC = () => {
   const [currentTabId, setCurrentTabId] = useState<number>();
   const [isRecordingSession, setIsRecordingSession] = useState<boolean>();
   const [isExplicitRecordingSession, setIsExplicitRecordingSession] = useState<boolean>();
 
-  const startRecordingOnClick = () => {
+  const startRecordingOnClick = useCallback(() => {
     sendEvent(EVENT.START_RECORDING_CLICKED);
     chrome.tabs.sendMessage(currentTabId, { action: CLIENT_MESSAGES.START_RECORDING }, { frameId: 0 }, () =>
       setIsRecordingSession(true)
     );
-  };
+  }, [currentTabId]);
 
-  const viewRecordedSession = () => {
+  const viewRecordedSession = useCallback(() => {
     window.open(`${config.WEB_URL}/sessions/draft/${currentTabId}`, "_blank");
-  };
+  }, [currentTabId]);
 
-  const stopRecordingOnClick = () => {
-    if (isExplicitRecordingSession) {
-      sendEvent(EVENT.STOP_RECORDING_CLICKED);
-      chrome.tabs.sendMessage(currentTabId, { action: CLIENT_MESSAGES.STOP_RECORDING }, { frameId: 0 }, () => {
-        setIsRecordingSession(false);
+  const stopRecordingOnClick = useCallback(
```

Comment:
Can we rename this to `viewRecordingOnClick` as view recording is the common thing, arg decides if we stop the recording?